packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3238,13 +3239,15 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   /// Returns the semantics that this node would like to add to its parent.
   _SemanticsFragment _getSemanticsForParent({
     required bool mergeIntoParent,
+    required bool blockUserInteractions,
   }) {
     assert(!_needsLayout, 'Updated layout information required for $this to calculate semantics.');
 
     final SemanticsConfiguration config = _semanticsConfiguration;
     bool dropSemanticsOfPreviousSiblings = config.isBlockingSemanticsOfPreviouslyPaintedNodes;
 
     final bool producesForkingFragment = !config.hasBeenAnnotated && !config.isSemanticBoundary;
+    final bool childrenBlockUserInteractions = blockUserInteractions || config.isBlockingUserInteractions;
```

Comment:
nit: I believe this is kinda the other way around, right? If this is true, interactions of children will be blocked. Maybe rename to `blockChildInteractions`?